pipenv/project.py

```diff
@@ -103,7 +103,7 @@ def requirements_exists(self):
     @property
     def virtualenv_exists(self):
         # TODO: Decouple project from existence of Pipfile.
-        if self.pipfile_exists:
+        if self.pipfile_exists and os.path.exists(self.virtualenv_location):
```

Comment:
I might be missing something here but what does this catch that the exists call in the return statement doesn't?

Reply:
This: https://github.com/kennethreitz/pipenv/issues/969#issuecomment-342704601

Reply:
Thanks @jck, I see the issue now. This looks sensible to me. I think that we really need to rethink how a lot of the project functions work because they've been pretty mangled together. We need clear cut cases of when we return  `''` vs `None`, and what they mean.

Reply:
@nateprewitt agreed but also better docstribgs could help for example.  It’s not totally clear to me why we return an empty string here which is why I didn’t understand this problem at all. But this check was safer and easier